src/Endpoints/AsyncSearch.php

```diff
@@ -35,11 +35,11 @@ class AsyncSearch extends AbstractEndpoint
 	 *
 	 * @param array{
 	 *     id: string, // (REQUIRED) The async search ID
-	 *     pretty: boolean, // Pretty format the returned JSON response. (DEFAULT: false)
-	 *     human: boolean, // Return human readable values for statistics. (DEFAULT: true)
-	 *     error_trace: boolean, // Include the stack trace of returned errors. (DEFAULT: false)
-	 *     source: string, // The URL-encoded request definition. Useful for libraries that do not accept a request body for non-POST requests.
-	 *     filter_path: list, // A comma-separated list of filters used to reduce the response.
+	 *     pretty?: bool, // Pretty format the returned JSON response. (DEFAULT: false)
+	 *     human?: bool, // Return human readable values for statistics. (DEFAULT: true)
+	 *     error_trace?: bool, // Include the stack trace of returned errors. (DEFAULT: false)
+	 *     source?: string, // The URL-encoded request definition. Useful for libraries that do not accept a request body for non-POST requests.
+	 *     filter_path?: string, // A comma-separated list of filters used to reduce the response.
```

Comment:
Note the code will actually also accept an array of sting elements: `string|array<string>`

Reply:
@AJenbo these array fields are specificed by the Elasticsearch rest-api-specification, reported [here](https://github.com/elastic/elasticsearch/tree/v8.17.3/rest-api-spec#type-definition). A `list` for this specification is represented as a comma separated list in the querystring, that is equivalent as a PHP string. That's why I used only `string` here.

Reply:
I understand that, but the implementation in this repo will take a list of strings and turn it in to a comma separated string:
https://github.com/elastic/elasticsearch-php/blob/603d87ed95fe47eee41bfaa32f855d38e8a89b80/src/Traits/EndpointTrait.php#L71

Not documenting this effectively turns this functionality in to dead code and prevents us from taking advantage of the convenience.

Reply:
@AJenbo unfortunately the `convertValue()` function that is able to translate an array<string> into a comma separated value is used only for the query string parameter (e.g. `filter_path`). I need to adjust the usage of `convertValue` also for other parameters in the URL, e.g. [here](https://github.com/elastic/elasticsearch-php/blob/main/src/Endpoints/AsyncSearch.php#L152). Anyway, you right we should use this feature with `string|array<string>`. Thanks for you help!